src/bentoml_io/models.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+import inspect
+import sys
+import typing as t
+from typing import ClassVar
+
+from pydantic import BaseModel
+from pydantic import Field
+from pydantic import RootModel
+from pydantic import create_model
+from typing_extensions import get_args
+
+from .typing_utils import is_file_like
+from .typing_utils import is_image_type
+from .typing_utils import is_iterator_type
+from .typing_utils import is_list_type
+
+if t.TYPE_CHECKING:
+    from starlette.requests import Request
+    from starlette.responses import Response
+
+    from .serde import Serde
+
+
+DEFAULT_STREAM_MEDIA_TYPE = "text/event-stream"
+DEFAULT_TEXT_MEDIA_TYPE = "text/plain"
+
+
+class IOMixin:
+    multipart_fields: ClassVar[list[str]]
+    media_type: ClassVar[str | None] = None
```

Comment:
```suggestion
    multipart_fields: ClassVar[t.List[str]]
    media_type: ClassVar[t.Optional[str]] = None
```

<img width="1561" alt="Screenshot 2023-11-08 at 03 32 08" src="https://github.com/bentoml/BentoML/assets/29749331/c34db885-af61-4c07-bdf4-86fc4642474c">

We need to be careful with this syntax for serialisation purposes 🤔